src/CidrValidator.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace ipl\Validator;
+
+use ipl\I18n\Translation;
+use ipl\Stdlib\Str;
+
+/**
+ * Validate a classless inter-domain routing (CIDR)
+ */
+class CidrValidator extends BaseValidator
+{
+    use Translation;
+
+    public function isValid($value): bool
+    {
+        $this->clearMessages();
+
+        $pieces = Str::trimSplit($value, '/');
+        if (count($pieces) !== 2) {
+            $this->addMessage(sprintf($this->translate('CIDR "%s" is in the wrong format'), $value));
+
+            return false;
+        }
+
+        list($address, $prefix) = $pieces;
+        $inaddr = @inet_pton($address);
+        if ($inaddr === false) {
+            $this->addMessage(sprintf($this->translate('CIDR "%s" contains an invalid IP-Address'), $value));
+
+            return false;
+        }
+
+        if (! is_numeric($prefix) || (int) $prefix < 0) {
```

Comment:
Since we're explicit with all the exception messages anyways I suggest the following:

```
if (! is_numeric(...)) {
    // must be a number
}

$prefix = (int) $prefix;
$maxPrefixLength = $IPv6 ? 128 : 32;

// Your previous if is a little complex as well since it can only be V4 or V6. So, if it's not V6 t's V4.
if ($prefix < 0 || $prefix > $maxPrefixLength) {
   //  Prefix length must be between 1 and %d for %s addresses
}